src/TreeView/TreeView.js

```diff
@@ -0,0 +1,471 @@
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import shortid from '../utils/shortId';
+import React, { Component } from 'react';
+
+export class TreeCol extends Component {
+    render() {
+        const {
+            children,
+            className,
+            ...rest
+        } = this.props;
+
+        const cellClassName = classnames(
+            'fd-tree__col',
+            className
+        );
+
+        return (
+            <div
+                {...rest}
+                className={cellClassName}>
+                {children}
+            </div>
+        );
+    }
+}
+
+TreeCol.displayName = 'TreeCol';
+
+TreeCol.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string
+};
+
+export class TreeHead extends Component {
+    render() {
+        const {
+            buttonProps,
+            children,
+            className,
+            isExpanded,
+            onExpandAll,
+            ...rest
+        } = this.props;
+
+        const headerClassName = classnames(
+            'fd-tree',
+            'fd-tree--header',
+            className
+        );
+
+        return (
+            <div {...rest} className={headerClassName}>
+                <div className='fd-tree__row fd-tree__row--header'>
+                    {
+                        React.Children.map(children, (child, index) => {
+                            const isFirstTreeCol = index === 0 && child.type && child.type.name === 'TreeCol';
```

Comment:
Should these types of checks now be using `child.type.displayName` instead of `child.type.name`?  If so, these exist in many of these components.

Reply:
I've seen examples of both. What do you think?

Reply:
I've seen both as well.  We have gone through the work of adding display names for the components so it _seems_ like that's the preferred route now.  I'm open to thoughts from others though.

Reply:
Using `displayName` in 2f74f50